### PR TITLE
Portal: Add container prop to allow rendering portal in a specific DOM node

### PR DESCRIPTION
**Pull request by denkristoffer**
This PR is an initial take on how to address https://github.com/reach/reach-ui/issues/87

Sometimes it'd be helpful to be able to render `<Dialog />` in other DOM nodes than `document.body`. This PR enables this by allowing the user to pass a `container` prop with a React ref pointing to the DOM node that they want to render the portal containing the `<Dialog />` in.

Edit: I forgot to handle `aria-hidden`.

### CODE REVIEW

File: packages/portal/src/index.js [hunk 1]
```diff
@@ -2,16 +2,28 @@ import React from "react";
 import ReactDOM from "react-dom";
 import Component from "@reach/component-component";
 
-let Portal = ({ children, type = "reach-portal" }) => (
+let Portal = ({
+  children,
+  container = document.body,
```
Comment (everdimension) [inline packages/portal/src/index.js]: This might break server-side rendering.

**Comment by ryanflorence** [comment]
My hesitation here is that in order to have dialogs on top of dialogs (even though it's a design sin, we still do it all the time) we need to ensure that the document order is correct, that the newest dialogs render later in the document than the ones before.

Can you expand further on this?

>  It's very important for apps that get integrated into other apps.

**Comment by everdimension** [comment]
> Can you expand further on this?

Yeah, basically I meant something like this:

screen shot 2019-01-04 at 15 26 12

It makes sense when your app is something like a charting solution or a visual builder of some sort (e.g. some UI that serves one isolated purpose, but also influences the external app). You could argue that putting such app in an `iframe` would solve the problem, but using an `iframe` is not always desirable.

You could also imagine some smaller widget for which it wouldn't make sense to block the whole page.

**Comment by ryanflorence** [comment]
Why would the modal not cover the "external website" nav? My understanding of a modal dialog is to interrupt the previous UI, not display them together.

**Comment by andreiglingeanu** [comment]
Is there anything we can do to make this prop happen? I see that the PR is a bit outdated but I'm more than happy to work towards getting it ready to be merged.

Now to describe a little bit the use case for this: I have fixed width sidebar in a dashboard and a main working area. What I'm trying to achieve is centering the overlays relative to the working area -- that'll look a bit cleaner from the design standpoint. Exactly what @everdimension described.

To address the dialogs stacking one on top of another: I'll make sure to always pass the same `container` prop for all overlays inside the app. That way they can happily stack, but cover only the working area of the app. Even more, dialogs stacking is very unintuitive for users, and everything that goes >1 level deep is bad, in my opinion.

**Pull request closed without merging**